doc/dependency-resolution.md

```diff
@@ -0,0 +1,98 @@
+## Dependency Resolution
+
+**Problem Statement**
```

Comment:
I've mentioned that a bit in 'Tracking Dependencies' further down. I'll look at moving it up to make it more prominent.